workers/generic-worker/mounts_test.go

```diff
@@ -421,6 +421,63 @@ func TestFileMountNoSHA256(t *testing.T) {
 	)
 }
 
+func TestFileMountWithCompression(t *testing.T) {
+	setup(t)
+	taskID := CreateArtifactFromFile(t, "compressed-file-mount.txt.gz", "public/build/compressed-file-mount.txt.gz")
+
+	// whether permission is granted to task user depends if running under windows or not
+	// and is independent of whether running as current user or not
+	granting, _ := grantingDenying(t, "file", t.Name())
+
+	// No cache on first pass
+	pass1 := append([]string{
+		`Downloading task ` + taskID + ` artifact public/build/compressed-file-mount.txt.gz to .*`,
+		`Downloaded 89 bytes with SHA256 a37856e8cd10250f76dc076bb03d380b16a870dec31f3461223f753124a4b28a from task ` + taskID + ` artifact public/build/compressed-file-mount.txt.gz to .*`,
+		`Content from task ` + taskID + ` artifact public/build/compressed-file-mount.txt.gz .* matches required SHA256 a37856e8cd10250f76dc076bb03d380b16a870dec31f3461223f753124a4b28a`,
+		`Creating directory .* with permissions 0700`,
+		`Decompressing gz file .* to .*` + t.Name(),
+	},
+		granting...,
+	)
```

Comment:
Would be nice to add `` `testing file mounts with compression!` `` here after `cat`ing the file (see comment below about specifying a custom `Payload` to include the `cat` command). This way, we are really sure it was decompressed and e.g. not just copied.

Reply:
I like this, the only issue is, the `LogTest()` helper checks specifically for the `[mounts]` prefix on this output. I'll add a check elsewhere, not in these `pass` variables.